CONTRIBUTING.md

```diff
@@ -16,35 +17,84 @@
 1. [Create a fork](https://guides.github.com/activities/forking/) of SplashKit.
 2. Clone the forked repository into your preferred location.
 
-   ```sh
+   sh
    git clone --recursive -j2 https://github.com/<username>/splashkit-core.git
-   ```
+   
 
-3. Build the test project:
+3. Initialise submodules if they weren't included during clone:
+
+   sh
+   git submodule update --init --recursive
+   
+
+4. Build the test project:
 
    - In a terminal application, go into the cloned directory of your SplashKit
      fork, and type:
 
-     ```sh
+     sh
      cd projects/cmake
      cmake .
      make
-     ```
+     
 
-4. Run the test program by executing
+   - For Apple Silicon (arm64) Macs, you may need to specify the architecture:
 
-   ```sh
+     sh
+     cd projects/cmake
+     # Clean any previous build files if you encounter errors
+     rm -rf CMakeFiles CMakeCache.txt
+     cmake . -DCMAKE_OSX_ARCHITECTURES=arm64
```

Comment:
We may or may not need to set this variable in the macOS presets. Hold off for now, but I'm just making a note of it in case testing finds that it's required.